### ISSUE TITLE
IPv6 address valid check ignores scope ID

### ISSUE THREAD

**Issue by TonyRL** [AdguardTeam/AdGuardHome#3022]
Have a question or an idea? Please search it [on our forum](https://github.com/AdguardTeam/AdGuardHome/discussions) to make sure it was not yet asked. If you cannot find what you had in mind, please [submit it here](https://github.com/AdguardTeam/AdGuardHome/discussions/new).

### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details

* **Version of AdGuard Home server:**
  * v0.106.0
* **How did you install AdGuard Home:**
  * Docker
* **How did you setup DNS configuration:**
  * Router
* **If it's a router or IoT, please write device model:**
  * pfSense on ESXi
* **CPU architecture:**
  * AMD64
* **Operating system and version:**
  * pfSense 2.5.1

### Expected Behavior

AdGuard Home should accept IPv6 address with scope ID and initialise normally.

### Actual Behavior

AdGuard Home initialisation stop at `dnsServer.Prepare: setting up resolvers: systemResolvers`

### Screenshots

<details><summary>Screenshot:</summary>

</details>

```
 2021/04/29 00:47:23 [info] AdGuard Home, version v0.106.0
 2021/04/29 00:47:23 [info] Upgrade yaml: 8 to 9
 2021/04/29 00:47:23 [info] Upgrade yaml: 9 to 10
 2021/04/29 00:47:23 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/29 00:47:23 [info] auth: initialized.  users:1  sessions:3
 2021/04/29 00:47:23 [info] number of certs - 1
 2021/04/29 00:47:23 [info] Initialize web module
 2021/04/29 00:47:23 [fatal] dnsServer.Prepare: setting up resolvers: systemResolvers: lookup test1619628443469047584.org on [fe80::1b0b:b8a6:e98a:cb58%br0]:53: parsing "fe80::1b0b:b8a6:e98a:cb58%br0": the passed string is not a valid IP address
```
### Additional Information

I have my DHCPv6 Server disabled and RA Router mode set as Assisted in my pfSense IPv6 settings. The IPv6 DNS servers (`fe80::1b0b:b8a6:e98a:cb58` and some others) are also entered there. 
`br0` is a custom network on my Docker host (Unraid).
The initialisation was smooth on `v0.105.2` and older releases. I spent some time switching different version tags and find out this issue has occurred since `v0.106.0-b.2`. 

Log of `v0.105.2` (Nothing went wrong):
<details>

```
 2021/04/28 16:54:54 [info] AdGuard Home, version v0.105.2, channel release, arch linux amd64
 2021/04/28 16:54:54 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/28 16:54:54 [info] Auth: initialized.  users:1  sessions:3
 2021/04/28 16:54:54 [info] number of certs - 1
 2021/04/28 16:54:54 [info] Initialize web module
 2021/04/28 16:54:55 [info] AdGuard Home is available on the following addresses:
 2021/04/28 16:54:55 [info] Go to https://**********.com
 2021/04/28 16:54:55 [info] Go to http://127.0.0.1:80
 2021/04/28 16:54:55 [info] Go to http://127.0.0.1:3001 (BETA)
 2021/04/28 16:54:55 [info] Go to http://[::1]:80
 2021/04/28 16:54:55 [info] Go to http://[::1]:3001 (BETA)
 2021/04/28 16:54:55 [info] Go to http://192.168.1.251:80
 2021/04/28 16:54:55 [info] Go to http://192.168.1.251:3001 (BETA)
 2021/04/28 16:54:55 [info] Go to http://[2404:**********]:80
 2021/04/28 16:54:55 [info] Go to http://[2404:**********]:3001 (BETA)
 2021/04/28 16:54:55 [info] Go to http://[2404:**********]:80
 2021/04/28 16:54:55 [info] Go to http://[2404:**********]:3001 (BETA)
 2021/04/28 16:55:01 [info] Starting the DNS proxy server
 2021/04/28 16:55:01 [info] The server is configured to refuse ANY requests
 2021/04/28 16:55:01 [info] DNS cache is enabled
 2021/04/28 16:55:01 [info] MaxGoroutines is set to 50
 2021/04/28 16:55:01 [info] Creating the UDP server socket
 2021/04/28 16:55:01 [info] Listening to udp://[::]:53
 2021/04/28 16:55:01 [info] Creating a TCP server socket
 2021/04/28 16:55:01 [info] Listening to tcp://[::]:53
 2021/04/28 16:55:01 [info] Creating a TLS server socket
 2021/04/28 16:55:01 [info] Listening to tls://[::]:853
 2021/04/28 16:55:01 [info] Creating a QUIC listener
 2021/04/28 16:55:01 [info] Listening to quic://[::]:784
 2021/04/28 16:55:01 [info] Entering the UDP listener loop on [::]:53
 2021/04/28 16:55:01 [info] Entering the DNS-over-QUIC listener loop on [::]:784
 2021/04/28 16:55:01 [info] Entering the tls listener loop on [::]:853
 2021/04/28 16:55:01 [info] Entering the tcp listener loop on [::]:53
```
</details>

Log of `v0.106.0-b.1` (Still nothing went wrong)

<details>

```
 2021/04/28 16:35:36 [info] AdGuard Home, version v0.106.0-b.1, channel beta, arch linux amd64
 2021/04/28 16:35:36 [info] Upgrade yaml: 7 to 8
 2021/04/28 16:35:36 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/28 16:35:36 [info] Auth: initialized.  users:1  sessions:3
 2021/04/28 16:35:36 [info] number of certs - 1
 2021/04/28 16:35:36 [info] Initialize web module
 2021/04/28 16:35:38 [info] AdGuard Home is available on the following addresses:
 2021/04/28 16:35:38 [info] Go to https://**********
 2021/04/28 16:35:38 [info] Go to http://127.0.0.1:80
 2021/04/28 16:35:38 [info] Go to http://127.0.0.1:3001 (BETA)
 2021/04/28 16:35:38 [info] Go to http://[::1]:80
 2021/04/28 16:35:38 [info] Go to http://[::1]:3001 (BETA)
 2021/04/28 16:35:38 [info] Go to http://192.168.1.251:80
 2021/04/28 16:35:38 [info] Go to http://192.168.1.251:3001 (BETA)
 2021/04/28 16:35:38 [info] Go to http://[2404:**********]:80
 2021/04/28 16:35:38 [info] Go to http://[2404:**********]:3001 (BETA)
 2021/04/28 16:35:45 [info] Starting the DNS proxy server
 2021/04/28 16:35:45 [info] The server is configured to refuse ANY requests
 2021/04/28 16:35:45 [info] DNS cache is enabled
 2021/04/28 16:35:45 [info] MaxGoroutines is set to 50
 2021/04/28 16:35:45 [info] Creating the UDP server socket
 2021/04/28 16:35:45 [info] Listening to udp://[::]:53
 2021/04/28 16:35:45 [info] Creating a TCP server socket
 2021/04/28 16:35:45 [info] Listening to tcp://[::]:53
 2021/04/28 16:35:45 [info] Creating a TLS server socket
 2021/04/28 16:35:45 [info] Listening to tls://[::]:853
 2021/04/28 16:35:45 [info] Creating a QUIC listener
 2021/04/28 16:35:45 [info] Listening to quic://[::]:784
 2021/04/28 16:35:45 [info] Entering the UDP listener loop on [::]:53
 2021/04/28 16:35:45 [info] Entering the tls listener loop on [::]:853
 2021/04/28 16:35:45 [info] Entering the tcp listener loop on [::]:53
 2021/04/28 16:35:45 [info] Entering the DNS-over-QUIC listener loop on [::]:784
```
</details>

Log of `v0.106.0-b.2` (Something went wrong from v`v0.106.0-b.1` to `v0.106.0-b.2`)

<details>

```
 2021/04/29 00:31:03 [info] AdGuard Home, version v0.106.0-b.2
 2021/04/29 00:31:03 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/29 00:31:03 [info] auth: initialized.  users:1  sessions:3
 2021/04/29 00:31:03 [info] number of certs - 1
 2021/04/29 00:31:03 [info] Initialize web module
 2021/04/29 00:31:04 [fatal] dnsServer.Prepare: setting up resolvers: systemResolvers: lookup test1619627463356613393.org on [fe80::1b0b:b8a6:e98a:cb58%br0]:53: parsing "fe80::1b0b:b8a6:e98a:cb58%br0": the passed string is not a valid IP address
Container stopped
```
</details>

Log of `v0.106.0-b.3`

<details>

```
 2021/04/29 01:12:49 [info] AdGuard Home, version v0.106.0-b.3
 2021/04/29 01:12:49 [info] Upgrade yaml: 8 to 9
 2021/04/29 01:12:49 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/29 01:12:49 [info] auth: initialized.  users:1  sessions:3
 2021/04/29 01:12:49 [info] number of certs - 1
 2021/04/29 01:12:49 [info] Initialize web module
 2021/04/29 01:12:49 [fatal] dnsServer.Prepare: setting up resolvers: systemResolvers: lookup test1619629969253076409.org on [fe80::1b0b:b8a6:e98a:cb58%br0]:53: parsing "fe80::1b0b:b8a6:e98a:cb58%br0": the passed string is not a valid IP address
Container stopped
```
</details>

Log of `v0.106.0-b.4`

<details>

```
 2021/04/29 01:13:58 [info] AdGuard Home, version v0.106.0-b.4
 2021/04/29 01:13:58 [info] Upgrade yaml: 9 to 10
 2021/04/29 01:13:58 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/29 01:13:58 [info] auth: initialized.  users:1  sessions:3
 2021/04/29 01:13:58 [info] number of certs - 1
 2021/04/29 01:13:58 [info] Initialize web module
 2021/04/29 01:13:58 [fatal] dnsServer.Prepare: setting up resolvers: systemResolvers: lookup test1619630038139649825.org on [fe80::1b0b:b8a6:e98a:cb58%br0]:53: parsing "fe80::1b0b:b8a6:e98a:cb58%br0": the passed string is not a valid IP address
```
</details>

Log of `v0.106.0-b.5`

<details>

```
 2021/04/29 01:15:24 [info] AdGuard Home, version v0.106.0-b.5
 2021/04/29 01:15:24 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/29 01:15:24 [info] auth: initialized.  users:1  sessions:3
 2021/04/29 01:15:24 [info] number of certs - 1
 2021/04/29 01:15:24 [info] Initialize web module
 2021/04/29 01:15:24 [fatal] dnsServer.Prepare: setting up resolvers: systemResolvers: lookup test1619630124160017595.org on [fe80::1b0b:b8a6:e98a:cb58%br0]:53: parsing "fe80::1b0b:b8a6:e98a:cb58%br0": the passed string is not a valid IP address
```
</details>

Log of `v0.106.0` (same as above)

<details>

```
 2021/04/29 00:47:23 [info] AdGuard Home, version v0.106.0
 2021/04/29 00:47:23 [info] Upgrade yaml: 8 to 9
 2021/04/29 00:47:23 [info] Upgrade yaml: 9 to 10
 2021/04/29 00:47:23 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/29 00:47:23 [info] auth: initialized.  users:1  sessions:3
 2021/04/29 00:47:23 [info] number of certs - 1
 2021/04/29 00:47:23 [info] Initialize web module
 2021/04/29 00:47:23 [fatal] dnsServer.Prepare: setting up resolvers: systemResolvers: lookup test1619628443469047584.org on [fe80::1b0b:b8a6:e98a:cb58%br0]:53: parsing "fe80::1b0b:b8a6:e98a:cb58%br0": the passed string is not a valid IP address
Container stopped
```
</details>

**Comment by TonyRL** [comment]
AdGuardHome.yaml
```yaml
bind_host: 0.0.0.0
bind_port: 80
beta_bind_port: 3001
users:
- name: *****
  password: *****
http_proxy: ""
language: zh-hk
rlimit_nofile: 0
debug_pprof: false
web_session_ttl: 720
dns:
  bind_host: 0.0.0.0
  port: 53
  statistics_interval: 90
  querylog_enabled: true
  querylog_file_enabled: true
  querylog_interval: 90
  querylog_size_memory: 1000
  anonymize_client_ip: false
  protection_enabled: true
  blocking_mode: default
  blocking_ipv4: ""
  blocking_ipv6: ""
  blocked_response_ttl: 10
  parental_block_host: family-block.dns.adguard.com
  safebrowsing_block_host: standard-block.dns.adguard.com
  ratelimit: 0
  ratelimit_whitelist: []
  refuse_any: true
  upstream_dns:
  - '[/168.192.in-addr.arpa/]192.168.1.1'
  - '[/local/]192.168.1.1'
  - '[/lan/]192.168.1.1'
  - '# Google'
  - sdns://AgUAAAAAAAAABzguOC44LjigHvYkz_9ea9O63fP92_3qVlRn43cpncfuZnUWbzAMwbkgdoAkR6AZkxo_AEMExT_cbBssN43Evo9zs5_ZyWnftEUKZG5zLmdvb2dsZQovZG5zLXF1ZXJ5
  - sdns://AgUAAAAAAAAAFlsyMDAxOjQ4NjA6NDg2MDo6ODg4OF2gHvYkz_9ea9O63fP92_3qVlRn43cpncfuZnUWbzAMwbkgdoAkR6AZkxo_AEMExT_cbBssN43Evo9zs5_ZyWnftEUKZG5zLmdvb2dsZQovZG5zLXF1ZXJ5
  - '#OpenDNS'
  - sdns://AQEAAAAAAAAADjIwOC42Ny4yMjAuMjIwILc1EUAgbyJdPivYItf9aR6hwzzI1maNDL4Ev6vKQ_t5GzIuZG5zY3J5cHQtY2VydC5vcGVuZG5zLmNvbQ
  - sdns://AQEAAAAAAAAAD1syNjIwOjA6Y2NjOjoyXSC3NRFAIG8iXT4r2CLX_WkeocM8yNZmjQy-BL-rykP7eRsyLmRuc2NyeXB0LWNlcnQub3BlbmRucy5jb20
  - '#Cloudflare'
  - sdns://AgcAAAAAAAAABzEuMC4wLjEAEmRucy5jbG91ZGZsYXJlLmNvbQovZG5zLXF1ZXJ5
  - sdns://AgcAAAAAAAAAFlsyNjA2OjQ3MDA6NDcwMDo6MTExMV0AIDFkb3QxZG90MWRvdDEuY2xvdWRmbGFyZS1kbnMuY29tCi9kbnMtcXVlcnk
  upstream_dns_file: ""
  bootstrap_dns:
  - 1.1.1.1
  - 1.0.0.1
  - 2606:4700:4700::1111
  - 2606:4700:4700::1001
  all_servers: true
  fastest_addr: false
  allowed_clients: []
  disallowed_clients: []
  blocked_hosts:
  - version.bind
  - id.server
  - hostname.bind
  cache_size: 4194304
  cache_ttl_min: 0
  cache_ttl_max: 0
  bogus_nxdomain: []
  aaaa_disabled: false
  enable_dnssec: false
  edns_client_subnet: true
  max_goroutines: 50
  ipset: []
  filtering_enabled: true
  filters_update_interval: 1
  parental_enabled: false
  safesearch_enabled: false
  safebrowsing_enabled: false
  safebrowsing_cache_size: 1048576
  safesearch_cache_size: 1048576
  parental_cache_size: 1048576
  cache_time: 30
  rewrites:
  - domain: *****
    answer: *****
  blocked_services: []
  customresolver: null
tls:
  enabled: true
  server_name: *****
  force_https: false
  port_https: 443
  port_dns_over_tls: 853
  port_dns_over_quic: 784
  port_dnscrypt: 0
  dnscrypt_config_file: ""
  allow_unencrypted_doh: false
  strict_sni_check: false
  certificate_chain: |-
    -----BEGIN CERTIFICATE-----
*****
    -----END CERTIFICATE-----
*****
    -----BEGIN EC PARAMETERS-----
*****
    -----END EC PARAMETERS-----
    -----BEGIN EC PRIVATE KEY-----
*****
    -----END EC PRIVATE KEY-----
  certificate_path: ""
  private_key_path: ""
filters:
- enabled: true
  url: https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt
  name: AdGuard DNS filter
  id: 1
- enabled: true
  url: https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts
  name: StevenBlack
  id: 1594206581
- enabled: false
  url: https://s3.amazonaws.com/lists.disconnect.me/simple_tracking.txt
  name: Disconnect.me Tracking
  id: 1594206582
- enabled: true
  url: https://s3.amazonaws.com/lists.disconnect.me/simple_ad.txt
  name: Disconnect.me Ads
  id: 1594206583
- enabled: true
  url: https://raw.githubusercontent.com/dead-hosts/cameleon_web_sysctl.org/master/origin.list
  name: Cameleon Hosts
  id: 1594206584
- enabled: true
  url: https://raw.githubusercontent.com/dead-hosts/hpHosts---Ad-and-Tracking_web_hosts-file.net/master/origin.list
  name: hpHosts
  id: 1594207929
- enabled: true
  url: https://pgl.yoyo.org/adservers/serverlist.php?hostformat=adblockplus&showintro=1&mimetype=plaintext
  name: Peter Lowe's List
  id: 1594207930
- enabled: true
  url: https://dbl.oisd.nl/
  name: oisd.nl
  id: 1594207932
- enabled: true
  url: https://raw.githubusercontent.com/TonyRL/blocklist/master/hosts
  name: Tony's blocklist
  id: 1594210473
- enabled: true
  url: https://someonewhocares.org/hosts/zero/hosts
  name: Dan Pollock's List
  id: 1594210474
- enabled: true
  url: https://raw.githubusercontent.com/notracking/hosts-blocklists/master/adblock/adblock.txt
  name: NoTracking
  id: 1596973928
- enabled: true
  url: https://www.github.developerdan.com/hosts/lists/ads-and-tracking-extended.txt
  name: Lightswitch05's ads-and-tracking-extended.txt
  id: 1602678200
- enabled: true
  url: https://adaway.org/hosts.txt
  name: AdAway Default Blocklist
  id: 1606134252
- enabled: true
  url: https://raw.githubusercontent.com/hoshsadiq/adblock-nocoin-list/master/hosts.txt
  name: NoCoin Filter List
  id: 1606134253
- enabled: true
  url: https://anti-ad.net/easylist.txt
  name: 'CHN: anti-AD'
  id: 1606134254
- enabled: true
  url: https://raw.githubusercontent.com/AdguardTeam/FiltersRegistry/master/filters/filter_224_Chinese/filter.txt
  name: Chinese filter
  id: 1606134255
whitelist_filters:
- enabled: true
  url: https://raw.githubusercontent.com/TonyRL/allowlist/master/domains
  name: Tony's allowlist
  id: 1594210472
user_rules:
- '*****'
dhcp:
  enabled: false
  interface_name: ""
  dhcpv4:
    gateway_ip: ""
    subnet_mask: ""
    range_start: ""
    range_end: ""
    lease_duration: 86400
    icmp_timeout_msec: 1000
    options: []
  dhcpv6:
    range_start: ""
    lease_duration: 86400
    ra_slaac_only: false
    ra_allow_slaac: false
clients:
- name: *****
  tags:
  - device_pc
  - os_windows
  - user_admin
  ids:
  - *****
  use_global_settings: true
  filtering_enabled: false
  parental_enabled: false
  safesearch_enabled: false
  safebrowsing_enabled: false
  use_global_blocked_services: true
  blocked_services: []
  upstreams: []
log_compress: false
log_localtime: false
log_max_backups: 0
log_max_size: 100
log_max_age: 3
log_file: ""
verbose: false
schema_version: 7

```

**Comment by ainar-g** [comment]
Thank you for the thorough report!  We'll investigate this.

**Comment by ainar-g** [comment]
Should be fixed on the edge channel as of snapshot f5adf15c.  Can you please check if our solution fixes the issue for you?

**Comment by TonyRL** [comment]
Yes. It fixed the issue. Thank you for your quick response.

Docker log
<details>

```
 2021/04/29 02:56:01 [info] AdGuard Home, version v0.107.0-a.2+f5adf15c
 2021/04/29 02:56:01 [info] Upgrade yaml: 7 to 8
 2021/04/29 02:56:01 [info] Upgrade yaml: 8 to 9
 2021/04/29 02:56:01 [info] Upgrade yaml: 9 to 10
 2021/04/29 02:56:01 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
 2021/04/29 02:56:01 [info] auth: initialized.  users:1  sessions:3
 2021/04/29 02:56:01 [info] number of certs - 1
 2021/04/29 02:56:01 [info] Initialize web module
 2021/04/29 02:56:04 [info] AdGuard Home is available on the following addresses:
 2021/04/29 02:56:04 [info] Go to https://******
 2021/04/29 02:56:04 [info] Go to http://127.0.0.1:80
 2021/04/29 02:56:04 [info] Go to http://127.0.0.1:3001 (BETA)
 2021/04/29 02:56:04 [info] Go to http://[::1]:80
 2021/04/29 02:56:04 [info] Go to http://[::1]:3001 (BETA)
 2021/04/29 02:56:04 [info] Go to http://192.168.1.251:80
 2021/04/29 02:56:04 [info] Go to http://192.168.1.251:3001 (BETA)
 2021/04/29 02:56:04 [info] Go to http://[2404:******]:80
 2021/04/29 02:56:04 [info] Go to http://[2404:******]:3001 (BETA)
 2021/04/29 02:56:04 [info] Go to http://[2404:******]:80
 2021/04/29 02:56:04 [info] Go to http://[2404:******]:3001 (BETA)
 2021/04/29 02:56:19 [info] Starting the DNS proxy server
 2021/04/29 02:56:19 [info] The server is configured to refuse ANY requests
 2021/04/29 02:56:19 [info] DNS cache is enabled
 2021/04/29 02:56:19 [info] MaxGoroutines is set to 50
 2021/04/29 02:56:19 [info] Creating the UDP server socket
 2021/04/29 02:56:19 [info] Listening to udp://[::]:53
 2021/04/29 02:56:19 [info] Creating a TCP server socket
 2021/04/29 02:56:19 [info] Listening to tcp://[::]:53
 2021/04/29 02:56:19 [info] Creating a TLS server socket
 2021/04/29 02:56:19 [info] Listening to tls://[::]:853
 2021/04/29 02:56:19 [info] Creating a QUIC listener
 2021/04/29 02:56:19 [info] Listening to quic://[::]:784
 2021/04/29 02:56:19 [info] Entering the tls listener loop on [::]:853
 2021/04/29 02:56:19 [info] Entering the UDP listener loop on [::]:53
 2021/04/29 02:56:19 [info] Entering the DNS-over-QUIC listener loop on [::]:784
 2021/04/29 02:56:19 [info] Entering the tcp listener loop on [::]:53
```

</details>
